src/features/vote/useVotingEvents.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from "react";
+import { ethers } from "ethers";
+import { MAINNET_DEPLOY_BLOCK } from "common/config";
+
+export interface PriceRound {
+  roundId: string;
+  identifier: string;
+  time: string;
+}
+
+const queryPriceRoundEvents = async (
+  contract: ethers.Contract,
+  setState: Function
+) => {
```

Comment:
nice pattern here, hoping we can generalize this even more. working on something similar with emp tools

Reply:
looking at this again, it could probably be simplified by just returning the state, rather than using the callback.  if there was a reason to call setState multiple times it would make more sense